init.py

```diff
@@ -300,68 +216,179 @@ def __():
+            title=mo.ui.text(
+                placeholder="The Elements of Marimo",
+                label="The title of your notebook: ",
+                full_width=True,
+            ),
+            path=mo.ui.text(
+                placeholder="marimo-tutorials/Computer-Science/..",
+                label="The path of your notebook: ",
+                full_width=True,
+            ),
+            tags=mo.ui.text(
+                placeholder="#deep-learning #transformers",
+                label="Tags of your notebook: ",
+                full_width=True,
+            ),
+            author_name=mo.ui.text(
+                placeholder="marimo-tutorials",
+                label="Your name: ",
+                full_width=True,
+            ),
+            author_contact=mo.ui.text(
+                placeholder="marimo-tutorials@example.com",
+                label="Your contact info: ",
+                full_width=True,
+            ),
+        )
+        .form(bordered=True, label="Custom Constants")
     )
-    return
+    return (init_form,)
 
 
-@app.cell
+@app.cell(hide_code=True)
 def __(init_form):
-    NOTEBOOK_TITLE = init_form.value["title"] if init_form.value else "Notebook Title"
+    NOTEBOOK_TITLE = (
+        init_form.value["title"] if init_form.value else "Notebook Title"
+    )
     NOTEBOOK_PATH = init_form.value["path"] if init_form.value else ".."
+    NOTEBOOK_TAGS = (
+        init_form.value["tags"]
+        if init_form.value
+        else "#deep-learning #transformers"
+    )
     AUTHOR_NAME = (
-        init_form.value["author_name"] if init_form.value else "marimo-tutorials-team"
+        init_form.value["author_name"]
+        if init_form.value
+        else "marimo-tutorials-team"
     )
     AUTHOR_CONTACT = (
         init_form.value["author_contact"]
         if init_form.value
         else "marimtutorial@example.com"
     )
-    return AUTHOR_CONTACT, AUTHOR_NAME, NOTEBOOK_PATH, NOTEBOOK_TITLE
-
-
-@app.cell
-def __(NOTEBOOK_TITLE, mo):
-    # implement sidebar in source code
-    sidebar_content = {
-        "#home": f"{mo.icon('lucide:home')} Home",
-        "#intro": "Introduction",
-        "Section 1": {
-            "#x": "Overview",
-            "#x-x": "Subsection x",
-        },  ## more sections below
-        "#summary": "Summary",
-        "#experiment": "Experiment",
-        "#refs": "References",
-        "#src": "Source Code",
-        "Links": {
-            "https://github.com/Haleshot/marimo-tutorials": f"{mo.icon('lucide:github')} GitHub",
-        },
-    }
-    mo.sidebar(
-        [
-            mo.md(f"# {NOTEBOOK_TITLE}"),
-            mo.nav_menu(sidebar_content, orientation="vertical"),
-        ]
+    return (
+        AUTHOR_CONTACT,
+        AUTHOR_NAME,
+        NOTEBOOK_PATH,
+        NOTEBOOK_TAGS,
+        NOTEBOOK_TITLE,
     )
-    return (sidebar_content,)
 
 
-@app.cell
+@app.cell(hide_code=True)
 def __(AUTHOR_CONTACT, AUTHOR_NAME, mo):
     mo.md(
         f"""
```

Comment:
We're including an Authors at the end of the notebook too?

Reply:
I think this is optional, we can remove this for now